src/troute-network/troute/DataAssimilation.py

```diff
@@ -687,6 +694,80 @@ def update_for_next_loop(self, network, da_run,):
         if not self.usgs_df.empty:
             self._usgs_df = self.usgs_df.loc[:,network.t0:]
 
+class great_lake(AbstractDA):
+    '''
+    Here is a list of the waterbody IDs and the gage they should correspond to:
+    4800002 -> 04127885 -> 
+    4800004 -> 04159130 -> 13196034 (segment_id)
+    4800006 -> 02HA013
+    4800007 -> IJC file    
+    '''
+    def __init__(self, network, from_files, value_dict, da_run):
+        
+        data_assimilation_parameters = self._data_assimilation_parameters
+        run_parameters = self._run_parameters
+        reservoir_persistence_da = data_assimilation_parameters.get('reservoir_da', None).get('reservoir_persistence_da', None)
+        
+        streamflow_da_parameters = data_assimilation_parameters.get('streamflow_da', None)
+        
+        if 'canada_timeslice_files' in da_run:
+            self._canada_df = _create_canada_df(data_assimilation_parameters, streamflow_da_parameters, run_parameters, network, da_run)
+        else:
+            self._canada_df = pd.DataFrame()
+        if 'LakeOntario_outflow' in da_run:
+            self._lake_ontario_df = _create_LakeOntario_df(run_parameters, network, da_run)
+        else:
+            self._lake_ontario_df = pd.DataFrame() 
+        # determine if user explictly requests streamflow DA
+        greatLake = False
+        if reservoir_persistence_da:
+            greatLake = reservoir_persistence_da.get('reservoir_persistence_greatLake', False)
+        
+        if greatLake:
+            lake_ontario_df = self._lake_ontario_df
+            canada_df = self._canada_df
+            
+            ids_to_check = {'4800002': '04127885', '4800004': '13196034'}
+            # the segment corresponding to 04127885 gages isn't exist as of now. Should be replaced in future
+            # Initialize an empty DataFrame with the same columns as the usgs DataFrame
+            if self._usgs_df.empty:
+                self._usgs_df = _create_usgs_df(data_assimilation_parameters, streamflow_da_parameters, run_parameters, network, da_run)
+            
+            usgs_df_GL = pd.DataFrame(columns=self._usgs_df.columns)
+
+            # Check if any ids are present in the index
+            for key, value in ids_to_check.items():
+                if value in self._usgs_df.index:
+                    temp_df = (self._usgs_df.loc[[value]]
+                            .transpose()
+                            .resample('15min')
+                            .asfreq()
+                            .transpose())
+                    temp_df.index = [key]
+                    usgs_df_GL = pd.concat([usgs_df_GL, temp_df], axis=0)
+                else:
+                    temp_df = pd.DataFrame(index=[key], columns=self._usgs_df.columns)
+                    usgs_df_GL = pd.concat([usgs_df_GL, temp_df], axis=0)   
```

Comment:
Aren't both `usgs_df_GL` and `temp_df` empty dataframes? Why join them, couldn't you just create an empty dataframe here with a specific index and columns?

Reply:
Because its in the for loop, `temp_df` checks if the values exist in the usgs_df if not add it in the usgs_df_GL.

Reply:
Done